### ISSUE TITLE
External link duplicating in portable text creating invalid HTML

### ISSUE THREAD

**Issue by CollierCZ** [kontent-ai/rich-text-resolver-js#48]
### Brief bug description

When I have a paragraph with two links, `transformToPortableText` adds both link marks to the second one. This makes the second link come out as nested `<a>` tags (which causes Svelte to yell at me 😞).

### Repro steps

```js
const testString = `<p>Text <a href="https://example.com">inner text 1</a> text between <a href="https://example.org">inner text 2</a>.</p>`
const parsedTest =  nodeParse(testString);
const portableTextTest = transformToPortableText(parsedTest);
console.log(parsedTest, portableTextTest)
```

The result is fine after parsing:

```json
[
  {
    "tagName": "p",
    "attributes": {},
    "children": [
      { "content": "Text ", "type": "text" },
      {
        "tagName": "a",
        "attributes": { "href": "https://example.com" },
        "children": [{ "content": "inner text 1", "type": "text" }],
        "type": "tag"
      },
      { "content": " text between ", "type": "text" },
      {
        "tagName": "a",
        "attributes": { "href": "https://example.org" },
        "children": [{ "content": "inner text 2", "type": "text" }],
        "type": "tag"
      },
      { "content": ".", "type": "text" }
    ],
    "type": "tag"
  }
]
```

But after `transformToPortableText`, the result comes out as:

```json
[
  { "_type": "span", "_key": "iSczJn1Pg8pTUjvX", "marks": [], "text": "Text " },
  {
    "_type": "span",
    "_key": "SWxS7iGMHitsRZom",
    "marks": ["iYc6g7G26JypEGko"],
    "text": "inner text 1"
  },
  { "_type": "span", "_key": "la7pr9mo2C8vgWZB", "marks": [], "text": " text between " },
  {
    "_type": "span",
    "_key": "Ec7Mp0px0yauHd8X",
    "marks": ["iYc6g7G26JypEGko", "NalzCVntF2xWEzzZ"],
    "text": "inner text 2"
  },
  { "_type": "span", "_key": "5byILNTDvtJohbog", "marks": [], "text": "." }
]
```

That's two marks on the second link.

For reference, the `markDefs`:

```json
[
  {
    "_key": "iYc6g7G26JypEGko",
    "_type": "link",
    "href": "https://example.com"
  },
  {
    "_key": "NalzCVntF2xWEzzZ",
    "_type": "link",
    "href": "https://example.org"
  }
]
```

### Expected behavior

The second link should have only one mark (`"NalzCVntF2xWEzzZ"` in the example).

### Test environment

 - Platform: Node 20
 - Version `"@kontent-ai/rich-text-resolver": "^1.0.1"`

**Comment by pokornyd** [comment]
hello Aaron, thanks for the detailed submission. I'll review the issue and update you on a solution, thanks for your patience.

**Comment by pokornyd** [comment]
this was a simple case of counter not reseting the links array, subsequently resulting in the overflow. fixed in release 1.1.2
